src/Microsoft.Health.Fhir.Shared.Api/Controllers/ImportController.cs

```diff
@@ -0,0 +1,203 @@
+            IUrlResolver urlResolver,
+            IOptions<OperationsConfiguration> operationsConfig,
+            IOptions<FeatureConfiguration> features,
+            ILogger<ImportController> logger)
+        {
+            EnsureArg.IsNotNull(fhirRequestContextAccessor, nameof(fhirRequestContextAccessor));
+            EnsureArg.IsNotNull(operationsConfig, nameof(operationsConfig));
+            EnsureArg.IsNotNull(urlResolver, nameof(urlResolver));
+            EnsureArg.IsNotNull(features?.Value, nameof(features));
+            EnsureArg.IsNotNull(mediator, nameof(mediator));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _fhirRequestContextAccessor = fhirRequestContextAccessor;
+            _importConfig = operationsConfig.Value.Import;
+            _urlResolver = urlResolver;
+            _features = features.Value;
+            _mediator = mediator;
+            _logger = logger;
+        }
+
+        [HttpPost]
+        [Route(KnownRoutes.Import)]
+        [ServiceFilter(typeof(ValidateImportRequestFilterAttribute))]
+        [AuditEventType(AuditEventSubType.Import)]
+        public async Task<IActionResult> Import([FromBody] Parameters importTaskParameters)
+        {
+            CheckIfImportIsEnabled();
+
+            ImportRequest importRequest = importTaskParameters.ExtractImportRequest();
+            ValidateImportRequestConfiguration(importRequest);
+
+            if (!ImportConstants.InitialLoadMode.Equals(importRequest.Mode, StringComparison.Ordinal))
+            {
+                throw new RequestNotValidException(Resources.OnlyInitialImportOperationSupported);
+            }
+
+            if (!importRequest.Force && !_importConfig.InitialImportMode)
+            {
+                throw new RequestNotValidException(Resources.InitialImportModeNotEnabled);
+            }
+
+            CreateImportResponse response = await _mediator.ImportAsync(
+                 _fhirRequestContextAccessor.RequestContext.Uri,
+                 importRequest.InputFormat,
+                 importRequest.InputSource,
+                 importRequest.Input,
+                 importRequest.StorageDetail,
+                 HttpContext.RequestAborted);
+
+            var bulkImportResult = ImportResult.Accepted();
+            bulkImportResult.SetContentLocationHeader(_urlResolver, OperationsConstants.Import, response.TaskId);
+            return bulkImportResult;
+        }
+
+        [HttpDelete]
+        [Route(KnownRoutes.ImportJobLocation, Name = RouteNames.CancelImport)]
+        [AuditEventType(AuditEventSubType.Import)]
+        public async Task<IActionResult> CancelImport(string idParameter)
+        {
+            CancelImportResponse response = await _mediator.CancelImportAsync(idParameter, HttpContext.RequestAborted);
+
+            _logger.LogInformation($"CancelImport {response?.StatusCode}");
+            return new ImportResult(response.StatusCode);
```

Comment:
same~ response?.StatusCode

Reply:
yes, we can add check here.

Reply:
The response should not be null, otherwise looks like deserve 500.